File: Data Structures & Algorithms/reorder-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public void reorderList(ListNode head) {
        ListNode slow = head;
        ListNode fast = head;
        while(fast!=null && fast.next!=null){
            slow = slow.next;
            fast = fast.next.next;
        }
        ListNode res = head;
        ListNode l2 = reverse(slow.next);
        slow.next = null;
        ListNode curr = head;
        while(l2!=null){
            ListNode l1n = curr.next;
            ListNode l2n = l2.next;
            curr.next = l2;
            l2.next = l1n;
            curr = l1n;
            l2 = l2n;
        }
        head = res;
    }

    public ListNode reverse(ListNode head){
        ListNode curr = head;
        ListNode prev = null;
        while(curr!=null){
            ListNode temp = curr.next;
            curr.next = prev;
            prev = curr;
            curr = temp;
        }
        return prev;
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        m = {}
        for sk in strs:
            temp = ''.join(sorted(sk))
            if m.get(temp) is None:
                m[temp] = [sk]
            else:
                m.get(temp).append(sk)
        res = []
        for sk in m.values():
            res.append(sk)
        return res
            
        